easybuild/tools/module_generator.py

```diff
@@ -238,18 +285,181 @@ def set_alias(self, key, value):
         # quotes are needed, to ensure smooth working of EBDEVEL* modulefiles
         return 'set-alias\t%s\t\t%s\n' % (key, quote_str(value))
 
-    def set_fake(self, fake):
-        """Determine whether this ModuleGenerator instance should generate fake modules."""
-        _log.debug("Updating fake for this ModuleGenerator instance to %s (was %s)" % (fake, self.fake))
-        self.fake = fake
-        # fake mode: set installpath to temporary dir
-        if self.fake:
-            self.tmpdir = tempfile.mkdtemp()
-            _log.debug("Fake mode: using %s (instead of %s)" % (self.tmpdir, self.module_path))
-            self.module_path = self.tmpdir
+
+class ModuleGeneratorLua(ModuleGenerator):
+    """
+    Class for generating Lua module files.
+    """
+    SYNTAX = LUA_SYNTAX
+
+    MODULE_SUFFIX = '.lua'
+
+    def __init__(self, *args, **kwargs):
+        """ModuleGeneratorLua constructor."""
+        super(ModuleGeneratorLua, self).__init__(*args, **kwargs)
+
+    def module_header(self):
+        """Return module header string."""
+        return ''
+
+    def comment(self, msg):
+        """Return string containing given message as a comment."""
+        return " -- %s\n" % msg
+
+    def get_description(self, conflict=True):
+        """
+        Generate a description.
+        """
+
+        description = "%s - Homepage: %s" % (self.app.cfg['description'], self.app.cfg['homepage'])
+
+        lines = [
+            "local pkg = {}",
+            "help = [["
+            "%(description)s"
+            "]]",
+            "whatis([[Name: %(name)s]])",
+            "whatis([[Version: %(version)s]])",
+            "whatis([[Description: %(description)s]])",
+            "whatis([[Homepage: %(homepage)s]])"
+            "whatis([[License: N/A ]])",
+            "whatis([[Keywords: Not set]])",
+            "",
+            "",
+            'pkg.root="%(installdir)s"',
+            "",
+            ]
+
+        #@todo check if this is really needed, imho Lmod doesnt need this at all.
+        if self.app.cfg['moduleloadnoconflict']:
+            lines.extend([
+             'if ( not isloaded("%(name)s/%(version)s")) then',
+             '  load("%(name)s/%(version)s")',
+             'end',
+             ])
+
+        elif conflict:
```

Comment:
Hi,

I obviously missed all the fun here, last month ;-)

What you describe it actually happens with HPCBIOS_Bioinfo, because it includes both BWA and Samtools; however, the shading here is intentional, if someone can keep the $PATH ordering right.

There is still a problem though with the `module swap` operation, because env-modules are not designed to protect you against. Here's where Lmod may come to the rescue: it has some mechanism of priorities to preserve ordering. Have a look at it, I may be just rumbling here!